ext/websocket/01_websocket.js

```diff
@@ -363,28 +365,31 @@
       } else if (this[_readyState] === OPEN) {
         this[_readyState] = CLOSING;
 
-        PromisePrototypeThen(
+        PromisePrototypeCatch(
           core.opAsync("op_ws_close", {
             rid: this[_rid],
             code,
             reason,
           }),
-          () => {
+          (err) => {
             this[_readyState] = CLOSED;
-            const event = new CloseEvent("close", {
-              wasClean: true,
-              code: code ?? 1005,
-              reason,
+
+            const errorEv = new ErrorEvent("error", {
```

Comment:
I don't know if this is WebIDL dictated stuff, but this sort of seems like a case where the new `cause` error property could be used?

Reply:
`cause` is for `Error`s, not `ErrorEvent`s.

Reply:
Ah, indeed. Sorry, my mistake.